init/modules/EESSI/2023.06.lua

```diff
@@ -17,56 +17,123 @@ local eessi_os_type = "linux"
+        -- x86_64/intel/skylake_avx512:x86_64/intel/haswell:x86_64/generic
+        for archdetect_filter_cpu in string.gmatch(archdetect_options, "([^" .. ":" .. "]+)") do
+            if isDir(pathJoin(eessi_prefix, "software", eessi_os_type, archdetect_filter_cpu, "software")) then
+                -- use x86_64/amd/zen3 for now when AMD Genoa (Zen4) CPU is detected,
+                -- since optimized software installations for Zen4 are a work-in-progress,
+                -- see https://gitlab.com/eessi/support/-/issues/37
+                if (archdetect_filter_cpu == "x86_64/amd/zen4" and not os.getenv("EESSI_SOFTWARE_SUBDIR_OVERRIDE") == "x86_64/amd/zen4") then
+                    archdetect_filter_cpu = "x86_64/amd/zen3"
+                    if mode() == "load" then
+                        LmodMessage("Sticking to " .. archdetect_filter_cpu .. " for now, since optimized installations for AMD Genoa (Zen4) are a work in progress.")
+                    end
                 end
+                eessiDebug("Selected archdetect CPU: " .. archdetect_filter_cpu)
+                return archdetect_filter_cpu
             end
-            return archdetect_filter_cpu
         end
+        LmodError("Software directory check for the detected architecture failed")
+    else
+        -- Still need to return something
+        return nil
+    end
+end
+function archdetect_accel()
+    local script = pathJoin(eessi_prefix, 'init', 'lmod_eessi_archdetect_wrapper_accel.sh')
+    -- for unload mode, we need to grab the value before it is unset
+    local archdetect_accel = os.getenv("EESSI_ACCEL_SUBDIR") or (os.getenv("EESSI_ACCELERATOR_TARGET_OVERRIDE") or "")
+    if not os.getenv("EESSI_ACCELERATOR_TARGET_OVERRIDE ") then
+        if convertToCanonical(LmodVersion()) < convertToCanonical("8.6") then
+            LmodError("Loading this modulefile requires using Lmod version >= 8.6, but you can export EESSI_ACCELERATOR_TARGET_OVERRIDE to the available accelerator architecture in the form of: accel/nvidia/cc80")
+        end
+        source_sh("bash", script)
     end
-    LmodError("Software directory check for the detected architecture failed")
+    archdetect_accel = os.getenv("EESSI_ACCEL_SUBDIR") or archdetect_accel
+    eessiDebug("Got archdetect accel option: " .. archdetect_accel)
+    return archdetect_accel
 end
 local archdetect = archdetect_cpu()
+local archdetect_accel = archdetect_accel()
 local eessi_cpu_family = archdetect:match("([^/]+)")
 local eessi_software_subdir = archdetect
 local eessi_eprefix = pathJoin(eessi_prefix, "compat", eessi_os_type, eessi_cpu_family)
 local eessi_software_path = pathJoin(eessi_prefix, "software", eessi_os_type, eessi_software_subdir)
-local eessi_module_path = pathJoin(eessi_software_path, "modules", "all")
+local eessi_modules_subdir = pathJoin("modules", "all")
+local eessi_module_path = pathJoin(eessi_software_path, eessi_modules_subdir)
```

Comment:
-- eessi_module_path is the location of the _CPU_ module files, e.g.,
-- /cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/modules/all